grails-app/controllers/org/pih/warehouse/shipping/ShipmentController.groovy

```diff
@@ -888,13 +888,13 @@ class ShipmentController {
 
 
     def saveComment() {
-        def shipmentInstance = Shipment.get(params.shipmentId)
-        def recipient = (params.recipientId) ? User.get(params.recipientId) : null
-        def comment = new Comment(comment: params.comment, sender: session.user, recipient: recipient)
-        if (shipmentInstance) {
-            shipmentInstance.addToComments(comment).save(flush: true)
-            flash.message = "${warehouse.message(code: 'shipping.addedCommentToShipment.message', args: [params.comment, shipmentInstance.name])}"
+        Shipment shipmentInstance = Shipment.get(params.shipmentId)
+        if (!shipmentInstance) {
+            throw new Exception("Could not locate shipment with ID " + params.shipmentId)
         }
+        User recipient = (params.recipientId) ? User.get(params.recipientId) : null
```

Comment:
I know it's been there and you might've not realized that, but I think `User.get(params.recipientId)` should be enough.... unless there is a possibility that user's id would be `null`..

Reply:
I did not pay much attention toit, I'll try to mak sure that recipient is not null and if everything will be fine I'll fix it to be just `User.get(recipientId)`